src/analyzing/CFG_build.ml

```diff
@@ -461,6 +463,36 @@ let build_cfg_of_unused_lambdas state previ nexti =
          | Some fdef -> cfg_lambda state previ nexti fdef);
   Hashtbl.clear state.unused_lambdas
 
+(*****************************************************************************)
+(* Marking notes *)
+(*****************************************************************************)
+
+let mark_at_exit_nodes cfg =
+  let rec loop nodei =
+    let node = cfg.CFG.graph#nodes#find nodei in
+    match node.n with
+    (* Visit ancestor for exit, noop, goto, and join nodes. *)
+    | Exit
+    | NOther (Noop _)
+    | NGoto _
+    | Join ->
+        CFG.predecessors cfg nodei |> List.iter (fun (predi, _) -> loop predi)
+    (* Certain instruction nodes may be implicitly returned. *)
```

Comment:
Probably should update this comment to apply to finding exit nodes, as opposed to implicit returns.